src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,185 @@
+import java.io.*;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+
+public class Parser {
+
+    private TaskList tasks;
+
+    /**
+     * Class constructor specifying the TaskList.
+     * @param tasks
+     */
+    public Parser(TaskList tasks) {
+        this.tasks = tasks;
+    }
+
+    /**
+     * Parses the specified String as required.
+     * @param str the String to be parsed.
+     * @throws DukeException
+     * @throws IOException
+     * @throws ClassNotFoundException
+     */
+    public void parse(String str) throws DukeException, IOException, ClassNotFoundException {
+
+        ArrayList<Task> store = this.tasks.getList();
+
+        if (str.equals("blah")) {
+            throw new DukeException("\u2639 OOPS!!! I'm sorry, but I don't know what that means :-(");
+        }
+
+        while (str != null) {
+            if (str.equals("list")) {
+                int counter = 1;
+                System.out.println("-------------------------");
+                for (Task task : store) {
+                    if (task.getDescription() != null) {
+                        System.out.println(counter + ". [" + task.getType() + "][" + task.getStatusIcon() + "] " + task.getDescription());
+                        counter++;
+                    }
+                }
+                System.out.println("-------------------------");
+                break;
+            } else if (str.contains("done")) {
+                if (str.length() == 4) {
+                    throw new DukeException("\u2639 OOPS!!! Please state what task has been completed.");
+                }
+                int num = Integer.parseInt(str.substring(5)) - 1;
+                Task curr = store.get(num);
+                curr.markAsDone();
+                System.out.println("-------------------------");
+                System.out.println("Nice! I've marked this task as done:");
+                System.out.println("  [" + curr.getStatusIcon() + "] " + curr.getDescription());
+                System.out.println("-------------------------");
+                break;
+            } else if (str.contains("todo")) {
+                try {
+                    if (str.length() == 4) {
+                        throw new DukeException("\u2639 OOPS!!! The description of a todo cannot be empty.");
+                    }
+                    ToDos curr = new ToDos(str.substring(5));
+                    store.add(curr);
+                    System.out.println("-------------------------");
+                    System.out.println("Got it. I've added this task:");
+                    System.out.println("  [" + curr.getType() + "][" + curr.getStatusIcon() + "] "
```

Comment:
This part is a little bit long, and maybe messy. I think we should avoid long methods by abstracting it (usage of polymorphism).